### DISCUSSION TITLE
Is `each` parallel?

### DISCUSSION THREAD

**Discussion by otakutyrant** [nushell/nushell#10628] [Q&A] [answered]
I want to execute `ls | each { |it| whisper --device cuda:0 --output_format srt $it }`, but I suddenly worry that if `each` is parallel. If so, then how should I execute the command by sequence? Because I have no enough GPU resource to hold such multiple parallel commands.

**Comment by panicbit** [comment, marked as answer]
`each` runs everything sequentially. `par-each` runs in parallel and also allows you to control how much it does (check its help for details).

**Reply by otakutyrant** [reply]
Thank you.
